### PR TITLE
Refactor getAllocatedFriendRequest to have out parameter

### PR DESCRIPTION
**Pull request by jmoguilevsky**
## What does this PR change?

Modifies the interface and usages, so that it now depends on a `TryGet` like implementation.

## How to test the changes?

1. Launch the explorer
2. ...

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at d111dde</samp>

This pull request refactors the `GetAllocatedFriendRequest` method in the `FriendsController` class and updates its callers and tests accordingly. The new method signature returns a boolean value and uses an out parameter to pass the friend request object. This improves the error handling and reporting of the friend request analytics and HUD controllers. It also enhances the performance and responsiveness of the HUD controllers by using cancellation tokens.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/ReportFriendRequestToAnalyticsExtensions.cs [hunk 1]
```diff
@@ -9,12 +10,26 @@ public static class ReportFriendRequestToAnalyticsExtensions
...
         {
-            FriendRequest request = friendsController.GetAllocatedFriendRequest(friendRequestId);
-            socialAnalytics.SendFriendRequestError(request?.From, request?.To,
+            bool wasFound = friendsController.GetAllocatedFriendRequest(friendRequestId, out FriendRequest request);
+
+            var description = e is FriendshipException fe
+                ? fe.ErrorCode.ToString()
+                : FriendRequestErrorCodes.Unknown.ToString();
+
+            if (!wasFound)
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/ReportFriendRequestToAnalyticsExtensions.cs:19]: Just a suggestion: if you moved `friendsController.GetAllocatedFriendRequest(friendRequestId, out FriendRequest request)` here you would not need a separate variable `wasFound`

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendRequestHUD/ReceivedFriendRequestHUDController.cs [hunk 2]
```diff
@@ -71,9 +71,9 @@ async UniTaskVoid ShowAsync(string friendRequestId, CancellationToken cancellati
             {
                 this.friendRequestId = friendRequestId;
 
-                FriendRequest friendRequest = friendsController.GetAllocatedFriendRequest(this.friendRequestId);
+                bool wasFound = friendsController.GetAllocatedFriendRequest(this.friendRequestId, out FriendRequest friendRequest);
 
-                if (friendRequest == null)
+                if (!wasFound)
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendRequestHUD/ReceivedFriendRequestHUDController.cs:76]: here as well you could merge two expressions and get rid of the variable

**Review by mikhail-dcl — COMMENTED** [review]
Thank you, overall looks good to me. I have a general suggestion for shortening the code: getting rid of a separate variable `wasFound` as it is not needed anywhere and you could check the result of `GetAllocatedFriendRequest` directly in abranch condition

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs [hunk 3]
```diff
@@ -441,15 +441,11 @@ public async UniTask<IReadOnlyList<FriendRequest>> GetFriendRequestsAsync(int se
         public void GetFriendsWithDirectMessages(string userNameOrId, int limit) =>
             apiBridge.GetFriendsWithDirectMessages(userNameOrId, limit, 0);
 
-        public FriendRequest GetAllocatedFriendRequest(string friendRequestId)
+        public bool GetAllocatedFriendRequest(string friendRequestId, out FriendRequest result)
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs:444]: This pattern `bool` => `out value` usually has a"Try" prefix in the method name, I'd suggest renaming to `TryGetAllocatedFriendRequest`.

**Review by AjimenezDCL — APPROVED** [review]
Looks good! left a small comment but approving regardless

**Pull request merged**